native/core/src/parquet/mod.rs

```diff
@@ -641,6 +640,8 @@ pub unsafe extern "system" fn Java_org_apache_comet_parquet_Native_initRecordBat
     session_timezone: jstring,
 ) -> jlong {
     try_unwrap_or_throw(&e, |mut env| unsafe {
+        let task_ctx = TaskContext::default();
```

Comment:
Both `TaskContext::default()` and `SessionContext::new()` get a `RuntimeEnv::default()`. So how does this change things?

Reply:
For this use case it is more like aligning with DataFusion. Feature wise is the same.
`TaskContext` is a state on top of the running ingle query, `SessionContext` is global state

Reply:
The difference here is that this is really an API called by the Java code. This path does not have a [PhysicalPlanner](https://github.com/apache/datafusion-comet/blob/3c1fc06c465f668800d747f3feb6295340c3329b/native/core/src/execution/planner.rs#L147C12-L147C27) and consequently does not have access to the Session or Task context used by a native operator. 
Also, not really sure if it makes a performance difference. But I'm fine with this change as long as it doesn't break anything.

Reply:
> although they both call `RuntimeEnv::default()` but both get a new object.

Exactly; which is why I'm wondering if the change makes any difference at all.

Reply:
Oh sorry, I seem to have misunderstood your previous comment.

Reply:
The `session_context` of comet operator plannar is customized to configure `memory_pool` and `batch_size`, while the parquet datasource exec does not use `memory_pool` but uses `batch_size`. So we may need to respect `COMET_BATCH_SIZE` here.

https://github.com/apache/datafusion-comet/blob/main/native/core/src/execution/jni_api.rs#L262-L293

https://github.com/apache/datafusion/blob/30c457a0837ff88d6537456d0573e39814c51fad/datafusion/datasource/src/file_scan_config.rs#L182

Reply:
filed  https://github.com/apache/datafusion-comet/issues/1571